packages/hash/frontend/src/components/hooks/blockProtocolFunctions/ontology/useBlockProtocolAggregateEntityTypes.ts

```diff
@@ -0,0 +1,57 @@
+import { useLazyQuery } from "@apollo/client";
+
+import { useCallback } from "react";
+import {
+  GetAllLatestEntityTypesQuery,
+  GetAllLatestEntityTypesQueryVariables,
+} from "../../../../graphql/apiTypes.gen";
+import { getAllLatestEntityTypesQuery } from "../../../../graphql/queries/ontology/entity-type.queries";
+import { AggregateEntityTypesMessageCallback } from "./ontology-types-shim";
+
+export const useBlockProtocolAggregateEntityTypes = (): {
+  aggregateEntityTypes: AggregateEntityTypesMessageCallback;
+} => {
+  const [aggregateFn] = useLazyQuery<
+    GetAllLatestEntityTypesQuery,
+    GetAllLatestEntityTypesQueryVariables
+  >(getAllLatestEntityTypesQuery);
+
+  const aggregateEntityTypes = useCallback<AggregateEntityTypesMessageCallback>(
+    async ({ data }) => {
+      if (!data) {
+        return {
+          errors: [
+            {
+              code: "INVALID_INPUT",
+              message: "'data' must be provided for aggregateEntityTypes",
+            },
+          ],
+        };
+      }
+
+      const response = await aggregateFn({
+        query: getAllLatestEntityTypesQuery,
```

Comment:
Do we want a @todo here?

Reply:
Addressed in 362495ae96a6af4cd56c0a07b6223b64399e702a